crates/env/src/call/call_builder.rs

```diff
@@ -521,26 +488,43 @@ where
     }
 }
 
-impl<E, Args, RetType> CallBuilder<E, Set<Call<E>>, Args, RetType>
+impl<E, CallType, Args, RetType> CallBuilder<E, Unset<CallType>, Args, RetType>
 where
     E: Environment,
 {
     /// Sets the `callee` for the current cross-contract call.
-    pub fn callee(self, callee: E::AccountId) -> Self {
-        let call_type = self.call_type.value();
+    pub fn callee(
+        self,
+        callee: E::AccountId,
+    ) -> CallBuilder<E, Set<Call<E>>, Args, RetType> {
         CallBuilder {
-            call_type: Set(Call {
-                callee: Some(callee),
-                gas_limit: call_type.gas_limit,
-                transferred_value: call_type.transferred_value,
-            }),
+            call_type: Set(Call::new(callee)),
             call_flags: self.call_flags,
             exec_input: self.exec_input,
             return_type: self.return_type,
             _phantom: Default::default(),
         }
     }
 
+    /// Sets the `code_hash` for the current cross-contract delegate call.
+    pub fn code_hash(
```

Comment:
So @ascjones I guess what you're suggesting is we rename this to `delegate`?

Reply:
Yes exactly, so by calling one method or the other we instantiate the builder type for the type of call we want.

Reply:
Actually maybe the choices are 

1. `callee` & `code_hash`
2. `call` & `delegate`

Reply:
Sure, but then we should rename the `callee` builder method to `call`

Reply:
:brain: :handshake: :brain:

Reply:
More like 🥜 🤝 🥜

Reply:
deez nuts